src/components/details.tsx

```diff
@@ -56,11 +56,21 @@ function DownloadBar(props: { torrent: Torrent }) {
         prefix = "Downloaded";
     }
 
+    const config = useContext(ConfigContext);
     const now = Math.floor(percent * 1000);
     const nowStr = `${prefix}: ${now / 10}%`;
+    const active = props.torrent.rateDownload > 0 || props.torrent.rateUpload > 0;
+    const variant = torrentProgressbarVariant(props, config, active)
```

Comment:
You can pass torrent directly to `torrentProgressbarVariant()` instead of wrapping it into `props` object. 
Also move the `active` calculation inside and make it return {animate, variant} object that you can expand directly into props of the component with `{...result}` construct.